### PR TITLE
Update fetch to utilize latest custom signals

### PR DESCRIPTION
**Pull request by tusharkhandelwal8**
When we update custom signals using setCustomSignals, the latest custom signals are not retrieved in the subsequent fetch. Currently, we are required to reload the app to fetch them.

Link to the bug: [Fetch uses stale custom signal values](https://b.corp.google.com/issues/381353888)

### CODE REVIEW

**Comment by karenyz** [comment]
can you add a test for the behavior you are fixing?

File: firebase-config/src/test/java/com/google/firebase/remoteconfig/internal/ConfigFetchHandlerTest.java [hunk 1]
```diff
@@ -766,6 +768,20 @@ public void getInfo_hitsThrottleLimit_throttledStatus() throws Exception {
...
+  @Test
+  public void customSignals_updated_onSubsequentFetch() throws Exception {
+    Map<String, String> customSignals =
+        ImmutableMap.of(
+            "subscription", "premium",
+            "age", "20");
+    sharedPrefsClient.setCustomSignals(customSignals);
+    fetchCallToHttpClientUpdatesClockAndReturnsConfig(firstFetchedContainer);
+
+    assertWithMessage("Fetch() failed!").that(fetchHandler.fetch().isSuccessful()).isTrue();
```
Comment (ashish-kothari) [inline firebase-config/src/test/java/com/google/firebase/remoteconfig/internal/ConfigFetchHandlerTest.java:780]: This assertion seems unrelated to the test case, can we remove it?

File: firebase-config/src/test/java/com/google/firebase/remoteconfig/internal/ConfigFetchHandlerTest.java [hunk 2]
```diff
@@ -959,10 +981,26 @@ private void verifyETags(@Nullable String requestETag, String responseETag) thro
...
             /* customHeaders= */ any(),
             /* firstOpenTime= */ any(),
-            /* currentTime= */ any());
+            /* currentTime= */ any(),
+            /* customSignals= */ any());
     assertThat(sharedPrefsClient.getLastFetchETag()).isEqualTo(responseETag);
   }
 
+  private void verifyCustomSignals(Map<String, String> customSignals) throws Exception {
+    verify(mockBackendFetchApiClient)
```
Comment (ashish-kothari) [inline firebase-config/src/test/java/com/google/firebase/remoteconfig/internal/ConfigFetchHandlerTest.java:990]: I would suggest that we assert that the custom signals in the fetch arguments is the same as what is set in the shared preferences (instead of passing it as an argument to this verify method). Optionally (even better IMO), inline this assertion in the test method itself.

**Review by ashish-kothari — APPROVED** [review]
Thanks for the fix, LGTM!

File: firebase-config/src/test/java/com/google/firebase/remoteconfig/internal/ConfigFetchHandlerTest.java (hunk 1 above)
Comment (tusharkhandelwal8) [reply firebase-config/src/test/java/com/google/firebase/remoteconfig/internal/ConfigFetchHandlerTest.java]: I have removed the assertion and am now calling fetch directly.

File: firebase-config/src/test/java/com/google/firebase/remoteconfig/internal/ConfigFetchHandlerTest.java (hunk 2 above)
Comment (tusharkhandelwal8) [reply firebase-config/src/test/java/com/google/firebase/remoteconfig/internal/ConfigFetchHandlerTest.java:989]: Updated it to assert that the custom signals in the fetch arguments is the same as what is set in the shared preferences.

File: firebase-config/src/test/java/com/google/firebase/remoteconfig/internal/ConfigFetchHandlerTest.java [hunk 3]
```diff
@@ -766,6 +768,19 @@ public void getInfo_hitsThrottleLimit_throttledStatus() throws Exception {
         .isEqualTo(firstFetchedContainer.getFetchTime());
   }
 
+  @Test
+  public void customSignals_updated_onSubsequentFetch() throws Exception {
```
Comment (karenyz) [inline firebase-config/src/test/java/com/google/firebase/remoteconfig/internal/ConfigFetchHandlerTest.java:772]: nit: fetch_usesLatestCustomSignals?

File: firebase-config/src/test/java/com/google/firebase/remoteconfig/internal/ConfigFetchHandlerTest.java [hunk 4]
```diff
@@ -766,6 +768,19 @@ public void getInfo_hitsThrottleLimit_throttledStatus() throws Exception {
...
+    Map<String, String> customSignals =
+        ImmutableMap.of(
+            "subscription", "premium",
+            "age", "20");
+    sharedPrefsClient.setCustomSignals(customSignals);
+    fetchCallToHttpClientUpdatesClockAndReturnsConfig(firstFetchedContainer);
+    fetchHandler.fetch();
+
+    verifyCustomSignals(customSignals);
```
Comment (karenyz) [inline firebase-config/src/test/java/com/google/firebase/remoteconfig/internal/ConfigFetchHandlerTest.java:781]: +1 to Ashish's comment, I would also expect to see the actual assertion in this test

**Review by karenyz — APPROVED** [review]
